vendor/bat-native-ledger/src/bat/ledger/internal/database/database_unblinded_token.cc

```diff
@@ -527,6 +545,146 @@ void DatabaseUnblindedToken::MarkRecordListAsSpent(
   ledger_->RunDBTransaction(std::move(transaction), transaction_callback);
 }
 
+void DatabaseUnblindedToken::MarkRecordListAsReserved(
+    const std::vector<std::string>& ids,
+    const std::string& redeem_id,
+    ledger::ResultCallback callback) {
+  if (ids.empty()) {
+    callback(ledger::Result::LEDGER_ERROR);
+    return;
+  }
+
+  auto transaction = ledger::DBTransaction::New();
+
+  const std::string id_values = GenerateStringInCase(ids);
+
+  std::string query = base::StringPrintf(
+      "UPDATE %s SET redeem_id = ?, reserved_at = ? "
+      "WHERE ( "
+        "SELECT COUNT(*) FROM %s "
+        "WHERE reserved_at = 0 AND token_id IN (%s) "
+      ") = ? AND token_id IN (%s)",
+      kTableName,
+      kTableName,
+      id_values.c_str(),
+      id_values.c_str());
+
+  auto command = ledger::DBCommand::New();
+  command->type = ledger::DBCommand::Type::RUN;
+  command->command = query;
+
+  BindString(command.get(), 0, redeem_id);
+  BindInt64(command.get(), 1, braveledger_time_util::GetCurrentTimeStamp());
+  BindInt64(command.get(), 2, ids.size());
+
+  transaction->commands.push_back(std::move(command));
+
+  query = base::StringPrintf(
+      "SELECT token_id FROM %s "
+      "WHERE reserved_at != 0 AND token_id IN (%s)",
+      kTableName,
+      id_values.c_str());
+
+  command = ledger::DBCommand::New();
+  command->type = ledger::DBCommand::Type::READ;
+  command->command = query;
+
+  transaction->commands.push_back(std::move(command));
+
+  auto transaction_callback =
+      std::bind(&DatabaseUnblindedToken::OnMarkRecordListAsReserved,
+          this,
+          _1,
+          ids.size(),
+          callback);
+
+  ledger_->RunDBTransaction(std::move(transaction), transaction_callback);
+}
+
+void DatabaseUnblindedToken::OnMarkRecordListAsReserved(
+    ledger::DBCommandResponsePtr response,
+    size_t expected_row_count,
+    ledger::ResultCallback callback) {
+  if (!response ||
+      response->status != ledger::DBCommandResponse::Status::RESPONSE_OK) {
+    callback(ledger::Result::LEDGER_ERROR);
+    return;
+  }
+
+  if (response->result->get_records().size() != expected_row_count) {
+    callback(ledger::Result::LEDGER_ERROR);
+    return;
+  }
+
+  callback(ledger::Result::LEDGER_OK);
+}
+
+void DatabaseUnblindedToken::MarkRecordListAsFree(
```

Comment:
nit: Maybe name it as `MarkRecordListAsSpendable` as we use spendable in other cases as well

Reply:
sorry for suggesting first free now spendable

Reply:
No problem, that's more consistent. Fixed.